server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.replication;
+
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.test.BackgroundIndexer;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchIntegTestCase.ClusterScope;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertHitCount;
+
+@ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
```

Comment:
Why is numDataNodes set to 0?

Reply:
This sets the number of nodes to start the test with. [ If unset a random number is used.](https://github.com/opensearch-project/OpenSearch/blob/d892c51d666536d48562c701691f40c1c5cafde0/test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java#L1719)